server/src/main/kotlin/org/itmodreamteam/myrest/server/model/restaurant/RestaurantTable.kt

```diff
@@ -41,4 +44,13 @@ class RestaurantTable() : JpaEntity() {
         this.numberOfSeats = numberOfSeats
         this._waiters = mutableSetOf()
     }
+
+    fun addWaiter(waiter: Waiter) {
+        if (waiter.restaurant != this.restaurant) {
+            throw UserException("Официант не является сотрудником ресторана")
```

Comment:
It looks like you put piece of domain logic in POJO. I think it's better to move restaurant equality check from dummy JPA entity to business-logic service which is responsible to assign waiter to table.